src/main/java/redis/clients/jedis/MultiNodePipelineBase.java

```diff
@@ -115,13 +93,22 @@ public void close() {
     }
   }
 
+  private ExecutorService getThreadPool() {
```

Comment:
Do you want the user to call `createAndUseDefaultThreadPool` to enable the thread pool? Before again, `Executors.newFixedThreadPool(MULTI_NODE_PIPELINE_SYNC_WORKERS)` is still used here, and, I don't think `newFixedThreadPool` is needed, we already have thread pool with custom parameters. Just define it as static.

Reply:
I want to keep the current behavior as the default behavior just so a large[1] pipeline doesn't cause many other pipelines to be waiting for a long time.

[1] with many connections (master nodes) and many commands to each node.

Reply:
> [1] with many connections (master nodes) and many commands to each node.

In this case users should use `setThreadPool` to set it themselves.

The time to create and destroy `FixedThreadPool` each time will be longer than the execution pipeline (the operating system needs to call multiple system calls to create and destroy threads)